gtsam/navigation/Gal3ImuEKF.h

```diff
@@ -0,0 +1,107 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  Gal3ImuEKF.h
+ * @brief Extended Kalman Filter for IMU-driven Gal3
+ *
+ * @date  September 2025
+ * @authors Scott Baker
+ */
+
+#pragma once
+
+#include <gtsam/navigation/LeftLinearEKF.h>  // Include the base class
+#include <gtsam/geometry/Gal3.h>
+#include <gtsam/navigation/PreintegrationParams.h>
+
+namespace gtsam {
+
+/// Specialized EKF for IMU-driven on Gal3
+class GTSAM_EXPORT Gal3ImuEKF : public LeftLinearEKF<Gal3> {
+ public:
+  using Base = LeftLinearEKF<Gal3>;
+  using TangentVector = typename Base::TangentVector;  // Vector9
+  using Jacobian = typename Base::Jacobian;            // 9x9
+  using Covariance = typename Base::Covariance;        // 9x9
+
+  /**
+   * Construct with initial state/covariance and preintegration params (for
+   * gravity and IMU covariances)
+   * @param X0 Initial Gal3.
+   * @param P0 Initial covariance in tangent space at X0.
+   * @param params Preintegration parameters providing gravity and options.
+   */
+  Gal3ImuEKF(const Gal3& X0, const Covariance& P0,
+                 const std::shared_ptr<PreintegrationParams>& params);
+
+  /// Calculate W (gravity-only left composition, world-frame increments)
+  /// Gal3:
+  /// [R, v, p
+  /// 0, 1, t -> W = [I, g*dt, 1/2 * g * dt^2
+  /// 0, 0, 1]        0, 1, dt
+  ///                 0, 0, 1]
+  static Gal3 Gravity(const Vector3& n_gravity, double dt) {
+    return {Rot3(), 0.5*n_gravity*dt*dt, n_gravity*dt, dt};
+  }
+
+  /// Calculate U from raw IMU (no gravity): body-frame increments
+  static Gal3 IMU(const Vector3& omega_b, const Vector3& f_b, double dt) {
+    return {Rot3::Expmap(omega_b * dt), f_b * (0.5 * dt * dt), f_b * dt, dt};
+  }
+
+  /**
```

Comment:
same for these comments. `BaseImuEKF` could be templated and just document these *abstract* methods, dynamics and predict. Then here you just declare them as override with one liner doxygen.